general/restrict-hostnames/rego/src_test.rego

```diff
@@ -1623,4 +1617,51 @@ test_ingress_allowed_no_host {
 
 	# Empty set means no violations
 	result == set()
-}
\ No newline at end of file
+}
+
+test_vs_not_allowed_no_host_not_allowed {
+	vs := {
+		"apiVersion": "admission.k8s.io/v1beta1",
+		"kind": "AdmissionReview",
+		"review": {
+			"kind": {
+				"group": "networking.istio.io",
+				"kind": "VirtualService",
+			},
+			"operation": "CREATE",
+			"userInfo": {
+				"groups": null,
+				"username": "alice",
+			},
+			"object": {
+				"metadata": {
+					"name": "test",
+					"namespace": "test",
+				},
+				"spec": {
+					"hosts": [""],
+					"http": [{"match": [{"uri": {"prefix": "/finance"}}, {"uri": {"exact": "/finance"}}, {"uri": {"regex": "^/finance"}}]}],
+					"tcp": [{"route": [{"destination": {"host": "finance.test.scv.cluster.local"}}]}],
+					"tls": [{"match": [{"sniHosts": ["finance.test.scv.cluster.local"]}]}],
+				},
+			},
+		},
+	}
+
+	namespaces := {"test": {
+		"apiVersion": "v1",
+		"kind": "Namespace",
+		"metadata": {
+			"annotations": {"ingress.statcan.gc.ca/allowed-hosts": `[{"host": "test.com","path":"/"}]`},
+			"name": "test",
+		},
+	}}
+
+	exemptions := [""]
```

Comment:
I think the test case is failing because of the exemption.

Reply:
Oooooh! That's a good catch... Does this mean that empty string is exempted? lol

Reply:
You are right! @MichaelPatsula for the win!

Reply:
```suggestion
	exemptions := []
```